pkg/api/registered/registered.go

```diff
@@ -141,6 +145,27 @@ func RegisteredVersionsForGroup(group string) (ret []unversioned.GroupVersion) {
 	return
 }
 
+// IsThirdPartyAPIGroupVersion returns true if the api version is a user-registered group/version.
+func IsThirdPartyAPIGroupVersion(gv unversioned.GroupVersion) bool {
+	for ix := range thirdPartyGroupVersions {
+		if thirdPartyGroupVersions[ix] == gv {
+			return true
+		}
+	}
+	return false
+}
+
+// SetThirdPartyAPIGroupVersions sets the list of third party versions, overwrites anything existing.
+func SetThirdPartyAPIGroupVersions(gvs []unversioned.GroupVersion) {
```

Comment:
I don't see any overwrite-protection here.  I don't want to allow a third party GroupVersion to "steal" a GroupVersion that's actually registered.